src/components/forms/profile/update-user-address-form.tsx

```diff
@@ -0,0 +1,136 @@
+    id: "",
+    sub: "",
+    street_address: "string",
+    locality: "string",
+    region: "string",
+    postal_code: "string",
+    country: "string",
+    kind: "string",
+    created_at: "string",
+    updated_at: "string",
+    street_address_2: "string",
+    primary: false,
+  });
+
+  React.useEffect(() => {
+    if (userAddress) {
+      setUpdatedAddress(userAddress);
+    }
+  }, [userAddress]);
+
+  return (
+    <>
+      <Form
+        formID={formID}
+        onSubmit={async () => {
+          setFormID(uuidv4());
+          return;
+        }}
+      >
+        <InputText
+          type="text"
+          name="street-address"
+          placeholder="Enter your street address"
+          value={updatedAddress.street_address}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              street_address: value,
+            });
+          }}
+          label="Street address *"
+        />
+        <InputText
+          type="text"
+          name="street-address2"
+          placeholder="Enter your complementary street address"
+          value={updatedAddress.street_address_2}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              street_address_2: value,
+            });
+          }}
+          label="Complementary street address"
+        />
+        <InputText
+          type="text"
+          name="locality"
+          placeholder="Enter your locality"
+          value={updatedAddress.locality}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              locality: value,
+            });
+          }}
+          label="Locality *"
+        />
+        <InputText
+          type="text"
+          name="locality"
+          placeholder="Enter your region"
+          value={updatedAddress.region}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              region: value,
+            });
+          }}
+          label="Region *"
+        />
+        <InputText
+          type="text"
+          name="locality"
+          placeholder="Enter your postal code"
+          value={updatedAddress.postal_code}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              postal_code: value,
+            });
+          }}
+          label="Postal code *"
+        />
+        <InputText
+          type="text"
+          name="locality"
+          placeholder="Enter your country"
+          value={updatedAddress.country}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              country: value,
+            });
+          }}
+          label="Country *"
+        />
+        <Button type="submit" variant={ButtonVariant.PRIMARY}>
+          Update my address
+        </Button>
```

Comment:
What do you think of already making the call to `PATCH /api/profile/addresses/:id` in this task?
Or is it only the front-end implementation?

Reply:
This is only the frontend  implementation :).

Reply:
Yup...
I just realised the task name was "CU-hh98q5-address_**form_page**" 😅

Reply:
Haha no worries, this is the place to ask *any* questions, or remarks :).